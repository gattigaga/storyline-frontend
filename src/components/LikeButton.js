import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import IconDefault from "react-icons/lib/io/android-favorite-outline";
import IconLiked from "react-icons/lib/io/android-favorite";

const Container = styled.div`
  width: 48px;
  height: 48px;
  border-radius: 100%;
  border: 1px solid ${({ isLiked }) => (isLiked ? "#e74c3c" : "#ccc")};
  display: flex;
  justify-content: center;
  align-items: center;
  background: white;
  cursor: pointer;
  transition: all 0.2s ease-in-out;

  &:hover {
    border: 1px solid #e74c3c;

    * {
      color: #e74c3c;
    }
  }
`;

const StyledIcon = styled(IconDefault)`
  font-size: 28px;
  color: #aaa;
  transition: all 0.2s ease-in-out;
  margin-top: 4px;
`;

const StyledIconLiked = StyledIcon.withComponent(IconLiked).extend`
  color: #e74c3c;
`;

const LikeButton = ({ isLiked, onClick }) => {
  return (
    <Container onClick={onClick} isLiked={isLiked}>
      {isLiked ? <StyledIconLiked /> : <StyledIcon />}
    </Container>
  );
};

LikeButton.propTypes = {
  isLiked: PropTypes.bool,
  onClick: PropTypes.func
};

export default LikeButton;
