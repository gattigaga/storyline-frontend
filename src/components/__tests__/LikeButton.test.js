import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import LikeButton from "../LikeButton";

describe("LikeButton", () => {
  const setup = propOverrides => {
    const props = {
      isLiked: false,
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<LikeButton {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders as liked", () => {
    const { wrapper } = setup({ isLiked: true });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
