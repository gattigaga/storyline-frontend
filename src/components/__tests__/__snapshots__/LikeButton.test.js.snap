// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton should renders as liked 1`] = `
<styled.div
  isLiked={true}
  onClick={[Function]}
>
  <Styled(IoAndroidFavorite) />
</styled.div>
`;

exports[`LikeButton should renders default 1`] = `
<styled.div
  isLiked={false}
  onClick={[Function]}
>
  <Styled(IoAndroidFavoriteOutline) />
</styled.div>
`;
